code/modules/mob/dead/observer/observer.dm

```diff
@@ -26,6 +26,13 @@ GLOBAL_VAR_INIT(observer_default_invisibility, INVISIBILITY_OBSERVER)
 	var/mob/observetarget = null	//The target mob that the ghost is observing. Used as a reference in logout()
 
 
+	/**
+	 * Are we in the ghost list?
+	 * 
+	 * This var is used so we can avoid adding multiple of this ghost to `GLOB.observer_list` whenever they reconnect
+	 */
+	var/ghost_listed = FALSE
```

Comment:
noted

+ I chose to keep AFK ghosts in the list so the bit in `data_huds.dm` for checking if the ghost is disconnected can work properly, I can't really think of a better way to do what I'm trying to do

Reply:
Basically the point is that the ghost list should be accurate, or at the very least it should be accurate as far as all this revive stuff is concerned. If something sets ghost_listed = FALSE, then that should also be removing them from the ghost list. If its setting it to true, it should be adding them to the list. If that is being done, then we don't need a var because we can just check the list and trust it (if its fucked due to OTHER code being shit, that's a separate issue to be fixed).